src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -659,17 +810,24 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
                             invokedAsExtensionMethod: false,
                             argsToParamsOpt: default(ImmutableArray<int>),
                             resultKind: LookupResultKind.Viable,
+                            binderOpt: null,
                             type: arrayEmpty.ReturnType);
                     }
                 }
             }
 
-            var int32Type = methodOrIndexer.ContainingAssembly.GetPrimitiveType(Microsoft.Cci.PrimitiveTypeCode.Int32);
+            return CreateParamArrayArgument(syntax, methodOrIndexer.ContainingAssembly, paramArrayType, arrayArgs, _compilation, _factory.CurrentMethod, _inExpressionLambda);
+        }
+
+        private static BoundExpression CreateParamArrayArgument(SyntaxNode syntax, AssemblySymbol containingAssembly, TypeSymbol paramArrayType, ImmutableArray<BoundExpression> arrayArgs, CSharpCompilation compilation, MethodSymbol currentMethod, bool inExpressionLambda)
+        {
+
+            var int32Type = containingAssembly.GetSpecialType(SpecialType.System_Int32);
 
             return new BoundArrayCreation(
                 syntax,
                 ImmutableArray.Create(
-                    MakeLiteral(syntax, ConstantValue.Create(arrayArgs.Length), int32Type)),
+                    MakeLiteral(syntax, ConstantValue.Create(arrayArgs.Length), int32Type, compilation, currentMethod, inExpressionLambda)),
```

Comment:
>MakeLiteral(syntax, ConstantValue.Create(arrayArgs.Length), int32Type, compilation, currentMethod, inExpressionLambda)) [](start = 20, length = 119)

It looks like we don't have to use MakeLiteral helper here. Let's create BoundLiteral node right here and remove no longer needed parameters.